index.js

```diff
@@ -1091,27 +1103,52 @@
           this._registerTrackers();
         }
 
-        // activates page tracking
-        if (trackRoutes) {
-          $rootScope.$on(pageEvent, function () {
-            // Apply $route based filtering if configured
-            if (readFromRoute) {
+        // This section adds different listeners based on the configuration
+        var eventListener = null;
+        if (typeof trackRouteFn === 'function') {
+          eventListener = function() {
+            // Redirect event to custom function, passing in protected object
+            trackRouteFn.apply(that, arguments);
+          };
+        } else if (trackRoutes) {
+          if (hybridMobileSupport) {
+            // For mobile devices we need to track a screen rather than a page
+            eventListener = function (event, state) {
+              that._send('screenview', {
```

Comment:
That's what I had in mind.
I haven't decided yet. I will setup a little test locally and see how
google analytics handles the missing field. I will also add a test to
ensure the values are present in Analytics.configuration.
Once it is done we should be good to go.